File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    var answer = 0;
     const frequencyMap = new Map();

    // 배열을 순회하면서 각 숫자의 등장 횟수를 저장
    for (let num of array) {
        frequencyMap.set(num, (frequencyMap.get(num) || 0) + 1);
    }

    let maxFrequency = 0;
    for (let [num, frequency] of frequencyMap.entries()) {
        if (frequency > maxFrequency) {
            maxFrequency = frequency;
            answer = num;
        } else if (frequency === maxFrequency) {
            answer = -1;
        }
    }
    return answer;
}

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.HashMap;
import java.util.Map;
// import java.util.*;

class Solution {
    public int solution(int[] array) {
        int answer = 0;
        Map<Integer, Integer> frequencyMap = new HashMap<>();

        for(int num : array) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        int maxFrequency = 0;
        for(Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            int frequency = entry.getValue();
            
            if(frequency > maxFrequency) {
                maxFrequency = frequency;
                answer = entry.getKey();
            }
            else if(frequency == maxFrequency) {
                answer = -1;
            }
        }
        return answer;
    }
}